src/ClientManager/Travely.ClientManager.Repository/Repository/BaseRepository.cs

```diff
@@ -0,0 +1,133 @@
+    {
+        protected TuristDbContext _dbContext;
+        public BaseRepository(TuristDbContext dbContext)
+        {
+            _dbContext = dbContext;
+
+        }
+
+        #region Base CRUD Operation
+
+        public void Add(T entity)
+        {
+            _dbContext.Set<T>().Add(entity);
+        }
+
+        public void Delete(T entity)
+        {
+            EntityEntry<T> dbEntityEntry = _dbContext.Entry(entity);
+
+            _dbContext.Set<T>().Attach(entity);
+
+            dbEntityEntry.State = EntityState.Deleted;
+        }
+
+        public IQueryable<T> Get(Expression<Func<T, bool>> predicate, params string[] includes)
+        {
+            IQueryable<T> set = _dbContext.Set<T>();
+
+            for (int i = 0; i < includes.Length; i++)
+            {
+                set = set.Include(includes[i]);
+            }
+
+            if (predicate == null)
+            {
+                return set;
+            }
+
+            return set.Where(predicate);
+        }
+
+        public IQueryable<T> Get(params string[] includes)
+        {
+            IQueryable<T> set = _dbContext.Set<T>();
+
+            for (int i = 0; i < includes.Length; i++)
+            {
+                set = set.Include(includes[i]);
+            }
+
+            return set;
+        }
+
+        public IQueryable<T> GetNoTracking(Expression<Func<T, bool>> predicate, params string[] includes)
+        {
+            IQueryable<T> set = _dbContext.Set<T>();
+
+            for (int i = 0; i < includes.Length; i++)
+            {
+                set = set.Include(includes[i]);
+            }
+
+            if (predicate == null)
+            {
+                return set;
+            }
+
+            return set.AsNoTracking().Where(predicate);
+        }
+
+        public IQueryable<T> GetNoTracking(params string[] includes)
+        {
+            IQueryable<T> set = _dbContext.Set<T>();
+
+            for (int i = 0; i < includes.Length; i++)
+            {
+                set = set.Include(includes[i]);
+            }
+
+            return set.AsNoTracking();
+        }
+
+        public void Update(T entity, params Expression<Func<T, object>>[] updatedProperties)
+        {
+            var dbEntityEntry = _dbContext.Entry(entity);
+
+            _dbContext.Set<T>().Attach(entity);
+
+
+            if (updatedProperties != null && updatedProperties.Any())
+            {
+                foreach (var property in updatedProperties)
+                {
+                    dbEntityEntry.Property(property).IsModified = true;
+                }
+            }
+            else
+            {
+                dbEntityEntry.State = EntityState.Modified;
+            }
+        }
+
+        #endregion
+
+        #region SaveChanges
+        public int SaveChanges()
+        {
+            return _dbContext.SaveChanges();
+        }
+
+        public async Task<int> SaveChangesAsync()
```

Comment:
No need to make method async , you can just return `_dbContext.SaveChangesAsync()` without awaiting it and await in relevant place